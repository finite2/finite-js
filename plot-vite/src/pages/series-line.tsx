import { LineSeriesExample0, LineSeriesExample1 } from "examples/series-line";

export const LineSeriesPage = () => {
  return (
    <>
      <LineSeriesExample0 />
      <LineSeriesExample1 />
    </>
  );
};
