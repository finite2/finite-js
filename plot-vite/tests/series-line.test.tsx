import * as LineExamples from "examples/series-line";
import { renderTest } from "./test-utils";

describe("LineSeries example render", () => {
  renderTest(LineExamples);
});
