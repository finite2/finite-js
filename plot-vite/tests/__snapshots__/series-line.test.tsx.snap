// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LineSeries example render > LineSeriesExample0 renders 1`] = `
<div
  id="test-id"
>
  <svg
    class="user-select-none relative"
    viewBox="0 0 900 600"
  >
    <g
      class="plot__xyplot"
    >
      <g
        class="plot__x-axis"
        transform="translate(50, 10)"
      >
        <line
          class="plot__axis-line"
          fill="var(--color-font)"
          stroke="var(--color-font)"
          x1="0"
          x2="840"
          y1="550"
          y2="550"
        />
        <g
          transform="translate(0, 550)"
        >
          <g
            transform="translate(0, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.0
            </text>
          </g>
          <g
            transform="translate(168, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.2
            </text>
          </g>
          <g
            transform="translate(336, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.4
            </text>
          </g>
          <g
            transform="translate(504, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.6
            </text>
          </g>
          <g
            transform="translate(672, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.8
            </text>
          </g>
          <g
            transform="translate(840, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              1.0
            </text>
          </g>
        </g>
      </g>
      <rect
        class="plot__draggable-axis cursor-e-resize"
        fill="#0000"
        height="40"
        width="840"
        x="50"
        y="560"
      />
      <g
        class="plot__y-axis"
        transform="translate(50, 10)"
      >
        <line
          class="plot__axis-line"
          fill="var(--color-font)"
          stroke="var(--color-font)"
          x1="0"
          x2="0"
          y1="0"
          y2="550"
        />
        <g
          transform="translate(0, 0)"
        >
          <g
            transform="translate(0, 550)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.0
            </text>
          </g>
          <g
            transform="translate(0, 440)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.2
            </text>
          </g>
          <g
            transform="translate(0, 330)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.4
            </text>
          </g>
          <g
            transform="translate(0, 220)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.6
            </text>
          </g>
          <g
            transform="translate(0, 109.99999999999997)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.8
            </text>
          </g>
          <g
            transform="translate(0, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              1.0
            </text>
          </g>
        </g>
      </g>
      <rect
        class="plot__draggable-axis cursor-s-resize"
        fill="#0000"
        height="550"
        width="50"
        x="0"
        y="10"
      />
      <g
        class="plot__series--line"
        transform="translate(50,10)"
      >
        <path
          d="M0,330L168,440L336,220L504,440L672,220L840,440"
          fill="#0000"
          stroke="var(--color-primary)"
          stroke-width="2"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`LineSeries example render > LineSeriesExample1 renders 1`] = `
<div
  id="test-id"
>
  <svg
    class="user-select-none relative"
    viewBox="0 0 900 600"
  >
    <g
      class="plot__xyplot"
    >
      <g
        class="plot__x-axis"
        transform="translate(50, 10)"
      >
        <line
          class="plot__axis-line"
          fill="var(--color-font)"
          stroke="var(--color-font)"
          x1="0"
          x2="840"
          y1="550"
          y2="550"
        />
        <g
          transform="translate(0, 550)"
        >
          <g
            transform="translate(0, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.0
            </text>
          </g>
          <g
            transform="translate(168, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.2
            </text>
          </g>
          <g
            transform="translate(336, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.4
            </text>
          </g>
          <g
            transform="translate(504, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.6
            </text>
          </g>
          <g
            transform="translate(672, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              0.8
            </text>
          </g>
          <g
            transform="translate(840, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="0"
              x2="0"
              y1="-6"
              y2="6"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.72em"
              text-anchor="middle"
              transform="translate(0, 14)"
            >
              1.0
            </text>
          </g>
        </g>
      </g>
      <rect
        class="plot__draggable-axis cursor-e-resize"
        fill="#0000"
        height="40"
        width="840"
        x="50"
        y="560"
      />
      <g
        class="plot__y-axis"
        transform="translate(50, 10)"
      >
        <line
          class="plot__axis-line"
          fill="var(--color-font)"
          stroke="var(--color-font)"
          x1="0"
          x2="0"
          y1="0"
          y2="550"
        />
        <g
          transform="translate(0, 0)"
        >
          <g
            transform="translate(0, 550)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.0
            </text>
          </g>
          <g
            transform="translate(0, 440)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.2
            </text>
          </g>
          <g
            transform="translate(0, 330)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.4
            </text>
          </g>
          <g
            transform="translate(0, 220)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.6
            </text>
          </g>
          <g
            transform="translate(0, 109.99999999999997)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              0.8
            </text>
          </g>
          <g
            transform="translate(0, 0)"
          >
            <line
              stroke="var(--color-font)"
              x1="6"
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              class="plot__tick-label select-none"
              dy="0.32em"
              text-anchor="end"
              transform="translate(-14, 0)"
            >
              1.0
            </text>
          </g>
        </g>
      </g>
      <rect
        class="plot__draggable-axis cursor-s-resize"
        fill="#0000"
        height="550"
        width="50"
        x="0"
        y="10"
      />
      <g
        class="plot__series--line"
        transform="translate(50,10)"
      >
        <path
          d="M0,110L168,220L336,275L504,220L672,110L840,55"
          fill="#0000"
          stroke="red"
          stroke-width="2"
        />
      </g>
      <g
        class="plot__series--line"
        transform="translate(50,10)"
      >
        <path
          d="M0,330L168,440L336,220L504,440L672,220L840,440"
          fill="#0000"
          stroke="blue"
          stroke-width="2"
        />
      </g>
    </g>
  </svg>
</div>
`;
